core/market/decentralized/src/matcher/matcher.rs

```diff
@@ -215,33 +215,23 @@ impl Matcher {
     pub async fn get_offer(
         &self,
         subscription_id: &SubscriptionId,
-    ) -> Result<Option<Offer>, MatcherError> {
-        let model_offer: Option<ModelOffer> = self
+    ) -> Result<Option<ModelOffer>, MatcherError> {
+        Ok(self
             .db
             .as_dao::<OfferDao>()
             .get_offer(subscription_id)
-            .await?;
-
-        match model_offer {
-            Some(model_offer) => Ok(Some(model_offer.into_client_offer()?)),
-            None => Ok(None),
-        }
+            .await?)
     }
 
     pub async fn get_demand(
         &self,
         subscription_id: &SubscriptionId,
-    ) -> Result<Option<Demand>, MatcherError> {
-        let model_demand: Option<ModelDemand> = self
+    ) -> Result<Option<ModelDemand>, MatcherError> {
+        Ok(self
             .db
             .as_dao::<DemandDao>()
             .get_demand(subscription_id)
-            .await?;
-
-        match model_demand {
-            Some(model_demand) => Ok(Some(model_demand.into_client_offer()?)),
-            None => Ok(None),
-        }
+            .await?)
```

Comment:
I'm not sure, if it is valuable change. Until now, these function could serve (in future) in production to get Offers. Now they are only for tests... But let's leave it

Reply:
They can be used. Just you need to append `.into_client_offer()` or `.into_client_demand()`